src/weathergen/utils/train_logger.py

```diff
@@ -377,17 +375,26 @@ def clean_df(df, columns: list[str] | None):
     return df
 
 
-def _clean_name(n: str) -> str:
-    """Cleans the stream name to only retain alphanumeric characters"""
-    return "".join([c for c in n if c.isalnum()])
+def clean_name(s):
+    """
+    Remove all characters from a string except letters, digits, and underscores.
+
+    Args:
+        s (str): The input string.
+
+    Returns:
+        str: A new string containing only alphanumeric characters and underscores,
+             in the same order and capitalization as they appeared in the input.
+    """
+    return "".join(c for c in s if c.isalnum() or c == "_")
 
 
 def _key_loss(st_name: str, lf_name: str) -> str:
-    st_name = _clean_name(st_name)
-    lf_name = _clean_name(lf_name)
+    st_name = clean_name(st_name)
+    lf_name = clean_name(lf_name)
```

Comment:
I think the loss function (`lf_name`) should remain untouched, right?

Reply:
The current clean_name wrapper keeps the alphanumeric and underscore characters so wrapping the lf_name shouldn't have any negative impact. Hence, left the original implementation that way.